src/runner/patch-math.js

```diff
@@ -0,0 +1,56 @@
+import {
+  BIG_DECIMAL,
+  DECIMAL_128,
+  PATCHED_MATH_METHODS,
+} from "../constants.js";
+
+const absImpl = {
+  [DECIMAL_128]: Decimal.abs.bind(Decimal),
+  [BIG_DECIMAL]: (arg) => arg.abs(),
+};
+
+const floorImpl = {
+  [DECIMAL_128]: Decimal.floor.bind(Decimal),
+  [BIG_DECIMAL](arg) {
+    const mode = arg.gt(0) ? Big.roundDown : Big.roundUp;
+    return arg.round(0, mode);
+  },
+};
+
+const log10Impl = {
+  [DECIMAL_128]: Decimal.log10.bind(Decimal),
+  [BIG_DECIMAL]() {
+    throwUnimplemented("Math.log10()", "BigDecimal");
+  },
+};
+
+const powImpl = {
+  [DECIMAL_128]: Decimal.pow.bind(Decimal),
+  [BIG_DECIMAL](base, exponent) {
+    return base.pow(exponent.toNumber());
+  },
+};
+
+const checkAndInitMathHandlers = (createUnaryHandler, createNaryHandler) => {
```

Comment:
Rather than leave this code unwrapped in `patches.js`, I think this check made it worth moving into its own file.